packages/matchbox/src/components/Slider/Slider.js

```diff
@@ -35,14 +40,35 @@ function Slider(props) {
   }, [value]);
 
   // Updates slider location when value changes
-  React.useLayoutEffect(() => {
-    const rect = getRectFor(sliderRef.current);
-    const absoluteProportion = (sliderValue + Math.abs(min)) / Math.abs(min - max);
-    setSliderLocation(lerp(0, rect.width, absoluteProportion));
-    if (onChange) {
-      onChange(sliderValue);
+  React.useEffect(() => {
+    if (rect.width) {
+      const absoluteProportion = (sliderValue + Math.abs(min)) / Math.abs(min - max);
+      setSliderLocation(lerp(0, rect.width, absoluteProportion));
+      if (onChange) {
+        onChange(sliderValue);
+      }
+    }
+  }, [sliderValue, rect.width]);
+
+  // Updates tick locations when ticks or window size change
+  const tickLocations = React.useMemo(() => {
+
+    if (!ticks || !rect.width) {
+      return {};
     }
-  }, [sliderValue, windowSize]);
+
+    return Object.keys(ticks).reduce((acc, number) => {
```

Comment:
Since we are going to map over tickLocations later, why not convert it here with map instead of reduce.

```
Object.keys(ticks).map((key) => {
  // ...
  return {
    position: lerp(0, rect.width, absoluteProportion),
    label: ticks[key],
    key: tick
  }
});
```

Reply:
Oh i completely overlooked this